barretenberg/cpp/src/barretenberg/client_ivc/client_ivc.cpp

```diff
@@ -283,32 +294,27 @@ void ClientIVC::complete_kernel_circuit_logic(ClientCircuit& circuit)
         instantiate_stdlib_verification_queue(circuit);
     }
 
+    bool is_init_kernel =
+        stdlib_verification_queue.size() == 1 && (stdlib_verification_queue.front().type == QUEUE_TYPE::OINK);
+
+    bool is_tail_kernel =
+        stdlib_verification_queue.size() == 1 && (stdlib_verification_queue.front().type == QUEUE_TYPE::PG_TAIL);
+
     bool is_hiding_kernel =
         stdlib_verification_queue.size() == 1 && (stdlib_verification_queue.front().type == QUEUE_TYPE::PG_FINAL);
 
-    // TODO(https://github.com/AztecProtocol/barretenberg/issues/1511): this should check the queue is of size one and
-    // contains an entry of type PG_TAIL, currently it might contain several entries in case it's a minimal transaction
-    // produced by our tests which is not exactly realistic
-    bool is_tail_kernel = std::any_of(stdlib_verification_queue.begin(),
-                                      stdlib_verification_queue.end(),
-                                      [](const auto& entry) { return entry.type == QUEUE_TYPE::PG_TAIL; });
-    // If the incoming circuit is a kernel, start its subtable with an eq and reset operation to ensure a
-    // neighbouring misconfigured subtable coming from an app cannot affect the operations in the
-    // current subtable. We don't do this for the hiding kernel as it succeeds another kernel.
-    if (!is_hiding_kernel) {
-        if (is_tail_kernel) {
-            // Add a no-op at the beginning of the tail kernel (the last circuit whose ecc ops subtable is prepended)
-            // to ensure the wires representing the op queue in translator circuit are shiftable polynomials, i.e.
-            // their 0th coefficient is 0.
-            circuit.queue_ecc_no_op();
-        }
-        circuit.queue_ecc_eq();
+    // The ECC-op subtable for a kernel begins with an eq-and-reset to ensure that the preceeding circuit's subtable
+    // cannot affect the ECC-op accumulator for the kernel. For the tail kernel, we additionally add a preceeding no-op
+    // to ensure the op queue wires in translator are shiftable, i.e. their 0th coefficient is 0. (The tail kernel
+    // subtable is at the top of the final aggregate table since it is the last to be prepended).
+    if (is_tail_kernel) {
+        circuit.queue_ecc_no_op();
     }
+    circuit.queue_ecc_eq();
```

Comment:
This is the reason I had to update the cached VKs. Previously we added this `eq` op to all circuits except the hiding. Its technically not needed for a kernel that always succeeds a kernel (so this also applies for the tail) but its simpler to just always add it.